br/pkg/lightning/restore/restore.go

```diff
@@ -972,7 +972,7 @@ func (rc *Controller) saveStatusCheckpoint(ctx context.Context, tableName string
 	switch {
 	case err == nil:
 		break
-	case utils.IsRetryableError(err), utils.MessageIsRetryableStorageError(err.Error()), common.IsContextCanceledError(err):
+	case utils.MessageIsRetryableStorageError(err.Error()), common.IsContextCanceledError(err):
```

Comment:
I agree with the plan to add a new function `IsRecoverableError`. Removing `MessageIsRetryableStorageError` can possibly fix this issue but some recoverable errors would be emitted either.

Reply:
Yes. After this PR, like DNS error would be emitted like before. Could you help add a `IsRecoverableError` later?

Reply:
> 

Sure. I'll file a new PR to fix it.